### PR TITLE
update arch tests for K extension

### PR DESCRIPTION
**Pull request by liweiwei90**
- update format for aes32 and sm4 instructions
- update reference signature for sha256 and sm3 instructions in rv64i_m/K_unratified
- delete zip and unzip tests in rv64i_m/K_unratified
- update tests for aes64ks1i, sm4ed and sm4ks to use byte_count with overlap = "Y" to improve the coverage of S-boxes

### CODE REVIEW

**Comment by ben-marshall** [comment]
@liweiwei90 - Thank you for this! Did you talk to IIT before doing this work / did you know they were already working on it?

@neelgala - I thought IIT Madras were working on these?

**Comment by liweiwei90** [comment]
Sorry, I didn't talk to IIT before. Our CAS/PLCT lab is the owner of two related task in the development partner(https://docs.google.com/spreadsheets/d/1qzu6b9kgADGjaa5fd1Qla7b9gCMOaEnGO5bUVu2oPys/edit#gid=156613795, line 26 and 27). I asked markku what we really need to do. He replied "_If all S-boxes and all the instruction invocations in a couple of encryptions, decryptions, and key schedules are covered, I think that is sufficient_" (the email was copied to you on Fri, Jul 30). I described the problems of s-boxes coverage I found in current arch tests in the last week meeting. This work is to  fix the problems. I also have sended two related PRs to riscv-ctg(https://github.com/riscv/riscv-ctg/pull/12) and riscv-isac(https://github.com/riscv/riscv-isac/pull/16) before this PR.

> @liweiwei90 - Thank you for this! Did you talk to IIT before doing this work / did you know they were already working on it?
> 
> @neelgala - I thought IIT Madras were working on these?

**Comment by neelgala** [comment]
Yes IIT has been working on this. I was notified about some issues in isac/ctg regarding these in the last developers meet - which I believe @liweiwei90 has fixed and raised PRs. There are additional changes required in the tests identified by IITM (the "K" in the RVTEST_CASE strings for Zks, Zkn, etc needs to be lower-cased everywhere for compatibility with the riscv-config yaml)

@liweiwei90 The artifacts of this PR were generated using your PRs on ctg and isac ? If so, I will merge those PRs, put in IIT's changes and then regenerate those artifacts. - would that work ?

**Comment by ben-marshall** [comment]
@liweiwei90 - which toolchain did you use to test this? I tried looking on the PLCT Jenkins server but it doesn't look like the GNU toolchain build is working or includes the latest spec changes?

Otherwise, these look like the changes I'd expect for the new spec updates, so however they get merged I'm happy about it.

**Comment by liweiwei90** [comment]
> Yes IIT has been working on this. I was notified about some issues in isac/ctg regarding these in the last developers meet - which I believe @liweiwei90 has fixed and raised PRs. There are additional changes required in the tests identified by IITM (the "K" in the RVTEST_CASE strings for Zks, Zkn, etc needs to be lower-cased everywhere for compatibility with the riscv-config yaml)
> 
> @liweiwei90 The artifacts of this PR were generated using your PRs on ctg and isac ? If so, I will merge those PRs, put in IIT's changes and then regenerate those artifacts. - would that work ?

@neelgala  yes, the *-01 tests are generated by riscv-ctg using the two PRs.  If you try to regenerate these artifacts, two changes should be made:
- Some *-rwp tests should be handwriting modified because of the format of the instructions have changed and they cannot be generated by riscv-ctg.
- Some reference signature should be updated since the function of some instructions have been changed to sign-extended the 32 bits result.

**Comment by liweiwei90** [comment]
> @liweiwei90 - which toolchain did you use to test this? I tried looking on the PLCT Jenkins server but it doesn't look like the GNU toolchain build is working or includes the latest spec changes?
> 
> Otherwise, these look like the changes I'd expect for the new spec updates, so however they get merged I'm happy about it.

@ben-marshall  Currently, I used the toolchain build from the riscv-crypto github following the steps described in it. I'll try to add it to our Jenkins server later. 
Since I used the latest version of the toolchain for the new spec, I found some errors when I built the old tests. So the modified tests in this patch are fit for the new spec.

File: riscv-test-suite/env/arch_test.h [hunk 1]
```diff
@@ -774,11 +774,11 @@ RVTEST_SIGUPD(swreg,destreg,offset)
     )
 
 //Tests for a instructions with register-register operand
-#define TEST_RI_OP(inst, destreg, reg2, imm, correctval, val1, val2, swreg, offset, testreg) \
+#define TEST_RI_OP(inst, destreg, reg1, reg2, imm, correctval, val1, val2, swreg, offset, testreg) \
```
Comment (pawks) [inline riscv-test-suite/env/arch_test.h:777]: Changing the definition for this macro affects the tests already present in the repository. If such a macro is needed, a new macro should be defined and used in the tests.

**Review by pawks — CHANGES_REQUESTED** [review]

File: riscv-test-suite/env/arch_test.h (hunk 1 above)
Comment (liweiwei90) [reply riscv-test-suite/env/arch_test.h:777]: All the instructions using  'TEST_RI_OP'  definition(aes32 and sm4 instructions) in the repo have been changed to the new format. So they should use the new 'TEST_RI_OP'  definition and the old 'TEST_RI_OP' will not be used after this change.

**Comment by liweiwei90** [comment]
> > @liweiwei90 - which toolchain did you use to test this? I tried looking on the PLCT Jenkins server but it doesn't look like the GNU toolchain build is working or includes the latest spec changes?
> > Otherwise, these look like the changes I'd expect for the new spec updates, so however they get merged I'm happy about it.
> 
> @ben-marshall Currently, I used the toolchain build from the riscv-crypto github following the steps described in it. I'll try to add it to our Jenkins server later.
> Since I used the latest version of the toolchain for the new spec, I found some errors when I built the old tests. So the modified tests in this patch are fit for the new spec.

@ben-marshall  the result can be found in the Jenkins server now: https://ci.rvperf.org/view/Krypto-Scalar/job/riscv-arch-test-k-extension/

**Comment by ben-marshall** [comment]
@liweiwei90 - That's great to see it working on your Jenkins server :+1:

**Comment by ben-marshall** [comment]
*Ping* - what needs to happen to move this forward?

**Pull request merged**